### ISSUE TITLE
Timers are stopped when scrolling due to scroll-observer

### ISSUE THREAD

**Issue by rsasum** [quasarframework/quasar#11998]
### What happened?

When scrolling timers that have been started via `setInterval` suddenly stop.

### What did you expect to happen?

Timers should not stop until clearTimeout is called.

### Reproduction URL

none

### How to reproduce?

1. Create a new quasar project.
2. Start a timer in `created` or `mounted` function of e.g. MainLayout and let it print some text.
3. Add some content to make it scrollable.
4. When started, the output of the timer can be seen and is updated in the console.
5. After scrolling a bit, no additional output is added. The timer has stopped.

### Flavour

Quasar CLI (@quasar/cli | @quasar/app)

### Areas

Components (quasar)

### Additional context

This is related to the scroll-observer (`quasar/src/components/scroll-observer/QScrollObserver.js`). In function `trigger`, the timer handle is set. This timer will be stopped via `clearTimout` in `emitEvent`. If props.debounce is false the timer handle will be set to the result of `requestAnimationFrame` which is not a timer handle. Thus random timers may be stopped.

```javascript
 function trigger (immediately) {
      if (immediately === true || props.debounce === 0 || props.debounce === '0') {
        emitEvent()
      }
      else if (timer === null) {
        timer = props.debounce
          ? setTimeout(emitEvent, props.debounce)
          : requestAnimationFrame(emitEvent) // <<<<<< Problem here, requestAnimationFrame does not return a timer handle
      }
    }
```

**Comment by pdanpdan** [comment]
:) nice one

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.4.8

**Comment by rsasum** [comment]
Thanx a lot 👍
